src/conn.rs

```diff
@@ -824,17 +791,33 @@ impl<const N: usize, P: ConnectionPeer> Connection<N, P> {
             self.readable.notify_one();
         }
 
-        // If both endpoints have exchanged FINs and all data has been received/acknowledged, then
-        // close the connection.
+        // One way Fin case 1, we are sending data, we send our fin and if it is acked, the reiecever got all our data and we close
+        if let State::Closing {
+            local_fin: Some(local),
+            remote_fin: None,
+            sent_packets,
+            ..
+        } = &mut self.state
+        {
+            if sent_packets.last_ack_num().is_some()
+                && sent_packets.last_ack_num().unwrap() == *local
+            {
+                self.state = State::Closed { err: None };
+            }
+        }
+
+        // One way Fin case 1, we are receiving data, we got the fin and sent an ack acked, close the connection
```

Comment:
`// One-way Fin: Case 2 - We have received data and then the fin. We sent an ack acked, so close the connection`